src/components/app/AppErrorBoundary.tsx

```diff
@@ -1,171 +1,214 @@
+      logInfo(`[AppErrorBoundary] routeError (ChunkLoadError): ${routeError}`);
+      openAppUpdateAvailableModal();
+      return;
     }
+    logError(`[AppErrorBoundary] routeError: ${routeError.message}`);
+  }, [routeError, openAppUpdateAvailableModal]);
+
+  return useMemo(() => ({
+    error: routeError,
+    isAppUpdateAvailable,
+  }), [routeError, isAppUpdateAvailable]);
+}
 
-    if (this.state.hasSuspenseError) {
-      // If there is a suspense error, wrap the children in
-      // a Suspense component with a fallback UI.
-      return (
-        <Suspense
-          fallback={(
-            <RouterFallback
-              loaderOptions={{
-                shouldCompleteBeforeUnmount: false,
-              }}
-            />
-          )}
-        >
-          {this.props.children}
-        </Suspense>
-      );
+function useHandleAsyncError() {
+  const asyncError = useAsyncError() as UseAsyncErrorResult;
+
+  useEffect(() => {
+    if (!asyncError) {
+      return;
     }
+    logError(`[AppErrorBoundary] asyncError: ${asyncError.message}`);
+  }, [asyncError]);
+
+  return asyncError;
+}
 
-    // If there is a non-suspense error, render an error page.
-    const {
-      title,
-      subtitle,
-      includeHelmet,
-      showSiteHeader,
-      showSiteFooter,
-      intl,
-    } = this.props;
+function useHandleErrorsOrAppUpdate(sourceError: Error | null = null) {
+  const {
+    error: routeError,
+    isAppUpdateAvailable,
+  } = useHandleRouteError();
+  const asyncError = useHandleAsyncError();
+
+  const error = sourceError || routeError || asyncError;
+  const errorMessageForDisplay = retrieveErrorMessageForDisplay(error);
+  const errorStackForDisplay = retrieveErrorStackForDisplay(error);
+
+  return useMemo(() => ({
+    error,
+    errorMessageForDisplay,
+    errorStackForDisplay,
+    isAppUpdateAvailable,
+  }), [error, errorMessageForDisplay, errorStackForDisplay, isAppUpdateAvailable]);
+}
+
+const AppErrorBoundary = ({
+  error = null,
+  title = null,
+  subtitle = null,
+  includeHelmet = false,
+  showSiteHeader = true,
+  showSiteFooter = true,
+}: AppErrorBoundaryProps) => {
+  const intl = useIntl();
+  const {
+    errorMessageForDisplay,
+    errorStackForDisplay,
+    isAppUpdateAvailable,
+  } = useHandleErrorsOrAppUpdate(error);
+  const { authenticatedUser }: AppContextValue = useContext(AppContext);
+
+  if (isAppUpdateAvailable) {
     return (
-      <ErrorPage
-        errorPageContentClassName="py-5 text-center"
-        title={title || intl.formatMessage(messages.errorTitle)}
-        subtitle={subtitle || intl.formatMessage(messages.errorSubtitle)}
-        includeHelmet={includeHelmet}
-        showSiteHeader={showSiteHeader}
-        showSiteFooter={showSiteFooter}
+      <AlertModal
+        title={intl.formatMessage(messages.updateAvailableModalHeading)}
+        variant="danger"
+        isOpen={isAppUpdateAvailable}
+        onClose={() => {}}
```

Comment:
[nit/non-blocking] Was this intentionally added to handle component prop requirements?

Reply:
Ah, good catch. The `onClose` prop for `AlertModal` is not required is `isBlocking` is provided, and defaults to `() => {}` anyways. Will remove.

<img width="578" alt="image" src="https://github.com/user-attachments/assets/a4969ee8-3da9-4aa8-a991-a6f92206a40b" />